data/config-schema/src/main/resources/net/corda/schema/configuration/crypto/1.0/corda.crypto.json

```diff
@@ -5,437 +5,195 @@
-      "type": "object",
-      "default": {},
-      "properties": {
-        "ops": {
-          "description": "Settings for crypto ops messages processor",
+        "maxAttempts": {
+          "description": "Maximum attempts to process a message",
           "type": "object",
-          "default": {},
           "properties": {
-            "maxAttempts": {
-              "description": "Maximum attempts to process a message",
+            "default": {
               "type": "integer",
               "default": 3
-            },
-            "waitBetweenMills": {
-              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
+            }
+          }
+        },
+        "waitBetweenMills": {
+          "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
+          "type": "object",
+          "properties": {
+            "default": {
               "type": "array",
+              "items": {
+                "type": "integer"
+              },
               "minItems": 1,
               "default": [
                 200
-              ],
-              "items": {
-                "type": "integer"
-              }
+              ]
             }
-          },
-          "additionalProperties": false
-        },
-        "flow": {
-          "description": "Settings for crypto flow ops messages processor",
+          }
+        }
+      }
+    },
+    "hsm": {
+      "description": "Settings and capabilities for the HSM",
+      "type": "object",
+      "default": {},
+      "properties": {
+        "retry": {
+          "description": "Retry settings for the HSM",
           "type": "object",
           "default": {},
           "properties": {
             "maxAttempts": {
-              "description": "Maximum attempts to process a message",
+              "description": "Maximum number of attempts",
               "type": "integer",
               "default": 3
             },
-            "waitBetweenMills": {
-              "description": "Time between attempts in milliseconds, if the number of values is less than attempts then the last item is repeated",
-              "type": "array",
-              "minItems": 1,
-              "default": [
-                200
-              ],
-              "items": {
-                "type": "integer"
-              }
+            "attemptTimeoutMills": {
+              "description": "Wait period in milliseconds between attempts, should be reasonably large as some operations may be long, e.g. RSA key generation by the SOFT HSM may take a few seconds",
+              "type": "integer",
+              "default": 20000
             }
           },
           "additionalProperties": false
         },
-        "registration": {
-          "description": "Settings for HSM registration messages processor",
+        "categories": {
```

Comment:
I think this could be removed (happy to be a follow-up PR).

Reply:
Yes will remove `categories` it was only meant for multi hsm concept.

Reply:
Removed in [#1047](https://github.com/corda/corda-api/pull/1047)